text/0001-catalogs.md

```diff
@@ -0,0 +1,341 @@
+# Catalogs: Shareable dependency version specifiers
+
+## Summary
+
+"_Catalogs_" allow multiple `package.json` files to share the same version specifier of a dependency through a new `catalog:` protocol.
+
+A catalog may be defined through JSON, JSON5, or YAML:
+
+```yaml
+# The schema version of the catalog file itself.
+catalog-definition-version: 1
+
+# Version specifiers that catalog consumers may reference.
+versions:
+  react: ^18.2.0
+  react-dom: ^18.2.0
+  redux: ^4.2.0
+  react-redux: ^8.0.0
+```
```

Comment:
I think we should make it simple ([related comment](https://github.com/orgs/pnpm/discussions/6376)). At least for now. I think we should only support catalogs through a new field in the workspace root `package.json`.

Loading catalogs from external packages can be added later but it could also be implemented by a third party system like `meta-updater`.

Reply:
named catalogs can still be supported if needed. But all the named catalogs would also be defined inline in the root `package.json`.

Reply:
That seems good to me! Implementing internal-only catalogs would be a good first implementation step either way. For what it's worth, this brings us back to the original version of this RFC, which proposed a root `pnpm.workspaceConsistent` config in `package.json`.

Is the preference for a new field on the workspace root `package.json` containing the full catalog strong? I think I came around to the idea of having a separate `catalog.json5` or `catalog.yaml` file since that makes it easier to add source code comments. How do you feel about only supporting `file:` in the first version and require that to be internal-only?

Reply:
pnpm supports `package.json5` and `package.yaml`

Reply:
For reasons similar to what was mentioned by others in https://github.com/pnpm/rfcs/pull/3, having `package.json` (named exactly) at the root can be valuable for other tools that don't understand `package.json5` and `package.yaml`.

Reply:
so we just need to update pnpm to generate a `package.json` from `package.json5`.

Reply:
I'm happy to simply stick the full catalog in the root `package.json` if there's a strong preference for that. 🙂 (This is what I originally proposed as well.)

> so we just need to update pnpm to generate a package.json from package.json5.

This tackles the read angle. There's a few tools I'm using that need to write to `package.json` and don't handle `package.json5`. For my specific use case, we'll probably just continue to avoid writing comments since switching to `package.json5` breaks other things for us. I suspect few others have tools that update the root `package.json`, so I'd be in favor of moving forward without that consideration.

Reply:
ok

Reply:
Cool! Updated here in commit 6334717f954f7045c9650a2196dc78232de125f8.